include/rmm/device_buffer.hpp

```diff
@@ -188,6 +191,7 @@ class device_buffer {
     other._size     = 0;
     other._capacity = 0;
     other.set_stream(cuda_stream_view{});
+    other._device = cuda_device_id{-1};
```

Comment:
question: Shouldn't we initialize `this->_device` with `other._device`?

Reply:
Yes!